plugin/gatsby-node.js

```diff
@@ -27,3 +27,171 @@ module.exports = runApisInSteps({
+    excludeFieldNames: Joi.array()
+      .items(Joi.string())
+      .description(
+        `A list of field names to globally exclude from the ingested schema.`
+      ),
+    html: Joi.object({
+      useGatsbyImage: Joi.boolean()
+        .default(true)
+        .description(
+          `Causes the source plugin to find/replace images in html with Gatsby images.`
+        ),
+      imageMaxWidth: Joi.number()
+        .integer()
+        .default(null)
+        .description(
+          `Adds a limit to the max width an image can be. If the image size selected in WP is smaller or the image file width is smaller than this those values will be used instead.`
+        ),
+      fallbackImageMaxWidth: Joi.number()
+        .integer()
+        .default(100)
+        .description(
+          `If a max width can't be inferred from html this value will be passed to Sharp. If the image is smaller than this, the image file's width will be used instead.`
+        ),
+      imageQuality: Joi.number()
+        .integer()
+        .default(90)
+        .description(
+          `Determines the image quality that Sharp will use when generating inline html image thumbnails.`
+        ),
+      createStaticFiles: Joi.boolean()
+        .default(true)
+        .description(
+          `When this is true, any url's which are wrapped in "", '', or () and which contain /wp-content/uploads will be transformed into static files and the url's will be rewritten. This adds support for , , and tags which point at WP media item uploads as well as inline-html css like background-image: url().`
+        ),
+    }).description(`Options related to html field processing.`),
+    type: Joi.object({
+      __all: getTypeOptions().description(
+        `A special type setting which is applied to all types in the ingested schema.`
+      ),
+      RootQuery: getTypeOptions().append({
+        excludeFieldNames: Joi.array()
+          .items(Joi.string())
+          .default([`viewer`, `node`, `schemaMd5`])
+          .description(`Excludes fields on a type by field name.`),
+      }),
+      MediaItem: Joi.object({
+        lazyNodes: Joi.boolean()
+          .default(false)
+          .description(
+            `Enables a different media item sourcing strategy. Instead of fetching Media Items that are referenced by other nodes, Media Items will be fetched in connection resolvers from other nodes. This may be desireable if you're not using all of the connected images in your WP instance. This is not currently recommended because it messes up cli output and can be slow due to query running concurrency.`
+          ),
+        localFile: Joi.object({
+          excludeByMimeTypes: Joi.array()
+            .items(`string`)
+            .default([])
+            .description(
+              `Allows preventing the download of files associated with MediaItem nodes by their mime types.`
+            ),
+        }),
+      }),
+    })
+      .pattern(Joi.string(), getTypeOptions())
```

Comment:
Aha! This must be the right way to handle what I've done here!

https://github.com/gatsbyjs/gatsby-source-wordpress-experimental/pull/229/files#diff-7a6e1ab692943ea7446ae0997cceb01fa9ea1f60e3482b0398d80ce4b46d8659R72